### ISSUE TITLE
Show last job status in UI in jobs list

### ISSUE THREAD

**Issue by trgtrg** [hashicorp/nomad#23893]
### Proposal

Show last run status - succeeded or failed.

### Use-cases

I have 100 periodic batch jobs and I want to see in one list if they all are working fine.
### Attempted Solutions

<img width="471" alt="failed or not" src="https://github.com/user-attachments/assets/5c476ba8-b653-4b70-914e-a826372402bc">

**Comment by philrenaud** [comment]
@trgtrg This is an awesome suggestion, thank you. I'll look into it, or if you or anybody else are interested, this would be a great issue to tackle.

**Comment by TheInvincibleRalph** [comment]
Hi, @philrenaud. I am interested in taking this but could you give more context as to what to do?

**Comment by philrenaud** [comment]
@TheInvincibleRalph Yeah! We did a pretty big overhaul to the jobs index page a few months ago: you can find @gulducat 's excellent work on it over at https://github.com/hashicorp/nomad/pull/20130 (and the UI parts at https://github.com/hashicorp/nomad/pull/20452)

You may note that there's a `ChildStatuses` property (https://github.com/hashicorp/nomad/pull/20130/files#diff-a3d5151e4786eb7bf28475ee38f9094f588edd49dd718c4f989d547902773eceR232-R250) for when a job has child jobs (if it's a parameterized or periodic job type)

This gives us a _little_ information, but not enough to make this feature really useful: this is because `Job.Status` gives us a pretty limited set of information: a job can be `Dead`, `Running`, or `Pending` (see https://github.com/hashicorp/nomad/blob/main/nomad/structs/structs.go#L4298-L4302)

The way we've started deriving more user-friendly statuses like "Complete" or "Failed", etc. is by looking that job's allocations: specifically if the job's total groups' `count` worth of allocations all have statuses that indicate things have gone well (see possible allocation statuses here: https://github.com/hashicorp/nomad/blob/main/nomad/structs/structs.go#L10903-L10910)

If one could do something like:
- get the latest job among child jobs for a given periodic/parameterized job,
- grab the allocations from that child job and surface them in a new property like `lastJobAllocations`, in the same format that you see for top-level jobs at the /statuses endpoint,

...then we could run them through the UI's [aggregateAllocStatus()](https://github.com/hashicorp/nomad/blob/main/ui/app/models/job.js#L217-L302) getter, which surfaces a user-friendly status that would be nice to append here.

(the final part of this would be to surface that information in the [jobs index table](https://github.com/hashicorp/nomad/tree/main/ui/app/templates/jobs/index.hbs#L174-L186) where this is some prior art for using Hds::Badges for this)

So: a couple steps, but this is a pretty neat feature with some fairly recent code paths!
